src/audio/drc/Testbench_drc_lin2dB/drc_lin2dB_fixed.c

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+
+//	drc_lin2dB_fixed.c :
+//	warning disabled to support file system
+//
+#pragma warning (disable : 4996)
+#pragma warning (disable : 4013)
+#include <stdio.h>
+#include "stdint.h"
+#include "math.h"
+#include "typdef.h"
+#include "norm.h"
+#include "testvector.h"
+#include <stdlib.h>
+#include <stddef.h>
+#include <string.h>
+int32_t drc_lin2db_fixed(int32_t linear, uint16_t idx, FILE* fd);		/* Input:Q6.26 Output:Q11.21 */
+static inline int32_t log10_fixed(int32_t x);
+
+
+
+int main()
+{
+	int32_t l_testvector, x[TEST_VECTOR], y[TEST_VECTOR]; 	/*Increase/decrease length as per need*/
+    uint16_t idx = 0;
+	int i;
+	/*int mkdir(const char* dirname);*/
+	mkdir("Results", 0777);
+	FILE* fd = fopen("Results/file.txt", "w");
+	/*fprintf(fd, " %10s| %10s|%10s|\n-----------+-----------+--------------+\n",  "idx", "testvector", "Fixlog10linear");*/
+	fprintf(fd, " %10s  %10s %10s \n", "idx", "testvector", "Fixlog10linear");
+
+    data_initialization_fixpt(x);
+    input_data_fixpt(x, y);
+	for (i = 0; i < TEST_VECTOR; i++)
+	{
+		l_testvector = y[i];
+		/* Data logging purpose_start*/
+        idx++;
+		fprintf(fd, " %10d %11li ", idx, l_testvector);
+		drc_lin2db_fixed(l_testvector, idx,fd);
+		/* Data logging purpose*/
+	}
+	fclose(fd);
+}
+
+/*
+ * function[y]= input_data_fixpt(x)
+ * Arguments    : const int32_t x[TEST_VECTOR]
+ *                int32_t y[TEST_VECTOR]
+ * Return Type  : void
+ */
+void input_data_fixpt(const int32_t x[TEST_VECTOR], int32_t y[TEST_VECTOR])
+{
+    memcpy(&y[0], &x[0], 639U * (sizeof(int32_t)));
+}
+
+/*
+ * Input is Q6.26: max 32.0
+ * Output range ~ (-inf, 30.1030); regulated to Q11.21: (-1024.0, 1024.0)
+ */
+
+inline int32_t drc_lin2db_fixed(int32_t linear, uint16_t idx, FILE* fd)		/* Data logging purpose*/
+{
+	int32_t log10_linear;
+	/* For negative or zero, just return a very small dB value. */
+	if (linear <= 0){
+		fprintf(fd, " %13.8f \n", -0.00047684);
+		return Q_CONVERT_FLOAT(-1000.0f, 21);
+	}
+	else{
+		log10_linear = log10_fixed(linear); /* Q6.26 */
+		log10_linear_log[idx] = log10_linear;		/* Dbg buffer*/
+		fprintf(fd, " %13li\n", log10_linear);
+		return q_mult(20, log10_linear, 0, 26, 21);	/*Q11.21*/ /*-67108866/2^21 = 32.0dB*/
+	}
+}
+/*
+ * Input depends on precision_x
+ * Output range [0.5, 1); regulated to Q2.30
+ */
+static inline int32_t rexp_fixed(int32_t x, int precision_x, int* e)
+{
+	int bit = 31 - norm_int32(x);
+
+	*e = bit - precision_x;
+
+	if (bit > 30)
+		return Q_SHIFT_RND(x, bit, 30);
+	if (bit < 30)
+		return Q_SHIFT_LEFT(x, bit, 30);
+	return x;
+}
+/*
+ * Input is Q6.26: max 32.0
+ * Output range ~ (-inf, 1.505); regulated to Q6.26: (-32.0, 32.0)
+ */
+static inline int32_t log10_fixed(int32_t x)
+{
+#define qc 26
```

Comment:
macro names are in upper case and need to be more descriptive - also best to always add an inline comment next to tuning macros too.